packages/xo-web/src/xo-app/disk-import/index.js

```diff
@@ -0,0 +1,188 @@
+            let type
+            if (
+              extIndex >= 0 &&
+              (type = name.slice(extIndex + 1)) &&
+              (type === 'vmdk' || type === 'vhd')
+            ) {
+              let vmdkData
+              if (type === 'vmdk') {
+                const parsed = await effects.parseVmdk(file)
+                vmdkData = {
+                  blocksTable: await parsed.tablePromise,
+                  capacity: parsed.capacityBytes,
+                }
+              }
+              push({
+                id: generateId(),
+                file,
+                name,
+                sr,
+                type,
+                vmdkData,
+              })
+            }
+          })
+        )
+        return { disks }
+      },
+      linkState,
+      onChangeDescription: (_, { target: { name, value } }) => ({
+        mapDescriptions,
+      }) => {
+        mapDescriptions[name] = value
+        return { mapDescriptions }
+      },
+      onChangeName: (_, { target: { name, value } }) => ({ mapNames }) => {
+        mapNames[name] = value
+        return { mapNames }
+      },
+      onChangeSr: (_, sr) => ({ sr }),
+      parseVmdk: async file =>
+        readCapacityAndGrainTable(async (start, end) => {
+          /* global FileReader */
+          const reader = new FileReader()
+          reader.readAsArrayBuffer(file.slice(start, end))
+          return (await fromEvent(reader, 'loadend')).target.result
+        }),
+      reset: getInitialState,
+    },
+  }),
+  injectIntl,
+  injectState,
+  class extends Component {
+    _import = () => {
+      const {
+        state: { disks, mapDescriptions, mapNames, sr },
+      } = this.props
+      return importDisks(
+        disks.map(disk => {
+          const { id, name } = disk
+          return {
+            ...disk,
+            name: mapNames[id] || name,
+            description: mapDescriptions[id],
+          }
+        }),
+        sr
+      )
+    }
+
+    render() {
+      const {
+        effects,
+        intl: { formatMessage },
+        state: { disks, mapDescriptions, mapNames, sr },
+      } = this.props
+      return (
+        <form id='import-form'>
+          <Row>
+            <LabelCol>{_('importToSr')}</LabelCol>
+            <InputCol>
+              <SelectSr onChange={effects.onChangeSr} required value={sr} />
+            </InputCol>
+          </Row>
+          {sr !== undefined && (
+            <div>
+              <Dropzone
+                onDrop={effects.handleDrop}
+                message={_('dropDisksFiles')}
+              />
+              {disks.length > 0 && (
+                <div>
+                  <div>
+                    {disks.map(({ file: { name, preview, size }, id }) => (
+                      <Collapse
+                        buttonText={`${name} - ${formatSize(size)}`}
+                        size='small'
+                        className='mb-1'
+                      >
+                        <br />
```

Comment:
Why?

Reply:
To have space betwwen collapsed button and the first input.